app_notifier/src/app_notifier/speech_notifier_plugin.py

```diff
@@ -52,6 +53,10 @@ def app_manager_stop_plugin(cls, app, ctx, plugin_args):
         notification = load_notification_jsons(json_paths)
         if 'object recognition' in notification:
             for event in notification['object recognition']:
+                start_date = datetime.datetime.fromtimestamp(
+                    self.start_time.secs).isoformat()
+                if event['date'] < start_date:
+                    continue
```

Comment:
```suggestion
                start_date = datetime.datetime.fromtimestamp(
                    self.start_time.to_sec())
                if dateutil.parser.isoparse(event['date']) < start_date:
                    continue
```

Reply:
please use `python-dateutil` and compare with `datetime.datetime`

```
In [17]: dateutil.parser.isoparse(datetime.datetime.fromtimestamp(rospy.Time.now().to_sec()).isoformat())
Out[17]: datetime.datetime(2021, 8, 3, 21, 11, 50, 620631)
```

Reply:
https://stackoverflow.com/questions/127803/how-do-i-parse-an-iso-8601-formatted-date

Reply:
Thank you for your advice!
I updated and tested mail_notifier.